crates/wasmcloud-host/src/actors/actor_host.rs

```diff
@@ -24,113 +23,284 @@ struct State {
+                image_ref: msg.image_ref,
+                host_id: hid,
+                seed: msg.signing_seed,
+                can_update: msg.can_update,
+            });
+            info!(
+                "Actor {} initialized",
+                &me.state.as_ref().unwrap().claims.subject
+            );
+            Ok(c.subject.to_string())
+        }
+        Err(_e) => {
+            error!(
+                "Failed to create a WebAssembly host for actor {}",
+                actor.token.claims.subject
+            );
+            ctx.stop();
+            Err("Failed to create a raw WebAssembly host".into())
+        }
+    }
+}
+
+fn advertise_claims(c: &Claims<wascap::jwt::Actor>, bus: &Addr<MessageBus>) -> bool {
+    let pc = AdvertiseClaims { claims: c.clone() };
+    block_on(async move {
+        if let Err(e) = bus.send(pc).await {
+            error!("Actor failed to advertise claims to bus: {}", e);
+            false
+        } else {
+            true
+        }
+    })
+}
+
+fn assert_validation_result(tv: &TokenValidation) -> Result<()> {
+    if tv.cannot_use_yet {
+        error!(
+            "Claims validation failure: Cannot be used {}",
+            tv.not_before_human
+        );
+        Err("Actor claims cannot be used yet".into())
+    } else if tv.expired {
+        error!("Claims validation failure: Expired {}", tv.expires_human);
+        Err("Actor claims have expired".into())
+    } else if !tv.signature_valid {
+        Err("Actor claims token has invalid signature".into())
+    } else {
+        Ok(())
+    }
+}
+
+fn validate_update(
+    new_claims: &Claims<wascap::jwt::Actor>,
+    old_claims: &Claims<wascap::jwt::Actor>,
+) -> Result<u32> {
+    if let Some(ref new_md) = new_claims.metadata {
+        if let Some(ref old_md) = old_claims.metadata {
+            if new_claims.subject != old_claims.subject {
                 error!(
-                    "Failed to create a WebAssembly host for actor {}",
-                    actor.token.claims.subject
+                    "Rejecting attempt to replace actor {} with actor {} - PKs do not match",
+                    old_claims.subject, new_claims.subject
                 );
-                ctx.stop();
-                Err("Failed to create a raw WebAssembly host".into())
+                return Err(
+                    "Public keys of old actor and new actor do not match. Update denied.".into(),
+                );
+            }
+            if new_md.rev.unwrap_or(0) <= old_md.rev.unwrap_or(0) {
+                return Err(
+                    "Cannot live update if the new module is not a higher revision number".into(),
+                );
+            }
+            if new_md.caps.as_ref().unwrap_or(&vec![]).len()
+                > old_md.caps.as_ref().unwrap_or(&vec![]).len()
+            {
+                warn!("Live update warning: new actor has more claims than the previous revision.");
             }
```

Comment:
I don't think we need it now, but I wonder if a `strict` mode can be enforced with a running host in the scenario where I'm managing my own host, and running 3rd party actors. In that case, I may want to deny a live update to an actor that decides to ask for more permissions. Thoughts?

Reply:
Also, I suppose, this statement doesn't check the contents of the claims, so an actor with `keyvalue` and `http_client` could ditch those, add in a `fs` capability and read files without triggering this warning. Could a `.iter().map()` be used in combination with https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.eq to achieve strict comparison?

Reply:
Right now I'm just sort of putting the warn in the logs so that someone can maybe check it out. We can always add issues in the future to strengthen the types of rules we have around live updates. We do also disable them by default

Reply:
I like the idea of having an issue to look into a "strict mode" for live update support